c7n/resources/vpc.py

```diff
@@ -993,7 +993,9 @@ def get_sg_refs(self):
             for perm_type in ('IpPermissions', 'IpPermissionsEgress'):
                 for p in sg.get(perm_type, []):
                     for g in p.get('UserIdGroupPairs', ()):
-                        sg_ids.add(g['GroupId'])
+                        # self references aren't usage.
+                        if g['GroupId'] != sg['GroupId']:
+                            sg_ids.add(g['GroupId'])
```

Comment:
odd I thought this was covered by existing tests, but seems like we need an explicit test cross references.